### DISCUSSION TITLE
[FEAT] RGB Colorcode support

### DISCUSSION THREAD

**Discussion by leodlpy7** [fastfetch-cli/fastfetch#1132] [Q&A]
## Wanted features:

The `keyColor`-attribute should support RGB-Colorcodes for a more colorful output.

## Motivation:

I am bored of the colors available and would like to customize them in a more flexible way using rgb. It would be great if you could implement the feature soon, I really miss it.

**Comment by CarterLi** [comment]
38;2;r;g;b

**Reply by TheDomphil** [reply]
so how do i set the keyColor with rgb value? 

Like this? "keyColor":"0;0;0;r;g;b"?

**Reply by IGrayfoxI** [reply]
> so how do i set the keyColor with rgb value?
> 
> Like this? "keyColor":"0;0;0;r;g;b"?

No.

38 means "Set Foreground Color"
The 2 after the 38 defines "RGB" mode.
the RGB values you want to set are directly after it.
So 38;2;255;128;0 means
Text is "Orange" with background unaffected.

You can add 1; before the 38 to enable "Bold" so 1;38;2;255;128;0 is Orange text that is more bold.
This wiki article explains what each number means
Note, some values are not supported.
https://en.wikipedia.org/wiki/ANSI_escape_code#Select_Graphic_Rendition_parameters.

Just a tip, some terminals do not support 24bit color and you might be limited to the 4bit color selection, you can also use the 8bit color selection by using 38;5;x
With x being the value of the color in the chart below that article from the link.
